programmers-coding-test/minjae/dfs-bfs/음료수얼려먹기.js

```diff
@@ -0,0 +1,50 @@
+const iceShape = [
+    [0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 0],
+    [1, 1, 1, 1, 1, 1, 0, 1, 1, 1, 1, 1, 1, 0],
+    [1, 1, 0, 1, 1, 1, 0, 1, 1, 0, 1, 1, 1, 0],
+    [1, 1, 0, 1, 1, 1, 0, 1, 1, 0, 0, 0, 0, 0],
+    [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+    [1, 1, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0],
+    [1, 1, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1],
+    [0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1],
+    [0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1],
+    [0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0],
+    [0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 0],
+    [0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0],
+    [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0, 0, 1, 1],
+    [1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1],
+    [1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1],
+];
+
+const ICE_SHAPE_X = iceShape.length;
+const ICE_SHAPE_Y = iceShape[0].length;
+
+//DFS는 큐1개,스택1개 or 재귀로 구현
+function solution() {
+    return allFreezeIceCube();
+}
+
+function freezeIceCube(x, y) {
+    if (x < 0 || y < 0 || x >= ICE_SHAPE_X || y >= ICE_SHAPE_Y) return false;
+    else if (iceShape[x][y] === 0) {
```

Comment:
해당 조건이 일치하면 return 으로 빠져나가므로 else if 구문이 필요없습니다!
```suggestion
    if (iceShape[x][y] === 0) {
```